pkg/config/config.go

```diff
@@ -3,15 +3,158 @@ package config
+	EnableCache   bool          `yaml:"enable_cache" json:"enable_cache"`
+	CacheTTL      time.Duration `yaml:"cache_ttl" json:"cache_ttl"`
+}
+
+// SecurityConfig represents security configuration
+type SecurityConfig struct {
+	ValidateSignatures bool     `yaml:"validate_signatures" json:"validate_signatures"`
+	AllowUnsignedRules bool     `yaml:"allow_unsigned_rules" json:"allow_unsigned_rules"`
+	TrustedSources     []string `yaml:"trusted_sources" json:"trusted_sources"`
+}
+
+// ReportingConfig represents reporting configuration
+type ReportingConfig struct {
+	IncludeMetadata    bool `yaml:"include_metadata" json:"include_metadata"`
+	IncludeRemediation bool `yaml:"include_remediation" json:"include_remediation"`
+	IncludeReferences  bool `yaml:"include_references" json:"include_references"`
+	GroupByCategory    bool `yaml:"group_by_category" json:"group_by_category"`
+	ShowProgress       bool `yaml:"show_progress" json:"show_progress"`
+}
+
+// IntegrationsConfig represents integrations configuration
+type IntegrationsConfig struct {
+	Webhooks []string `yaml:"webhooks" json:"webhooks"`
+	SARIF    struct {
+		IncludeRuleHelp  bool `yaml:"include_rule_help" json:"include_rule_help"`
+		IncludeLocations bool `yaml:"include_locations" json:"include_locations"`
+	} `yaml:"sarif" json:"sarif"`
+}
+
+// DevelopmentConfig represents development configuration
+type DevelopmentConfig struct {
+	Debug        bool `yaml:"debug" json:"debug"`
+	DryRun       bool `yaml:"dry_run" json:"dry_run"`
+	ValidateOnly bool `yaml:"validate_only" json:"validate_only"`
+	Experimental bool `yaml:"experimental" json:"experimental"`
+}
+
+// SpotterConfig represents the complete Spotter configuration
 type SpotterConfig struct {
+	// Legacy fields for backward compatibility
 	HubURL   string `yaml:"hub_url" json:"hub_url"`
 	APIKey   string `yaml:"api_key" json:"api_key"`
 	CacheDir string `yaml:"cache_dir" json:"cache_dir"`
 	RulesDir string `yaml:"rules_dir" json:"rules_dir"`
 	PacksDir string `yaml:"packs_dir" json:"packs_dir"`
+
+	// New comprehensive configuration sections
+	Logging      LoggingConfig      `yaml:"logging" json:"logging"`
+	Scanner      ScannerConfig      `yaml:"scanner" json:"scanner"`
+	Rules        RulesConfig        `yaml:"rules" json:"rules"`
+	Output       OutputConfig       `yaml:"output" json:"output"`
+	Kubernetes   KubernetesConfig   `yaml:"kubernetes" json:"kubernetes"`
+	Helm         HelmConfig         `yaml:"helm" json:"helm"`
+	Filtering    FilteringConfig    `yaml:"filtering" json:"filtering"`
+	Performance  PerformanceConfig  `yaml:"performance" json:"performance"`
+	Security     SecurityConfig     `yaml:"security" json:"security"`
+	Reporting    ReportingConfig    `yaml:"reporting" json:"reporting"`
+	Integrations IntegrationsConfig `yaml:"integrations" json:"integrations"`
+	Development  DevelopmentConfig  `yaml:"development" json:"development"`
 }
 
 // DefaultConfig returns the default configuration
```

Comment:
Just a small improvement point (for later), we can probably store a default config file right here in the repo and embed that into the binary

Reply:
Done